storage/redis/peer_store.go

```diff
@@ -263,32 +279,145 @@ type peerStore struct {
 	cfg Config
 	rb  *redisBackend
 
+	scripts scripts
+
 	closed chan struct{}
 	wg     sync.WaitGroup
 }
 
-func (ps *peerStore) groups() []string {
-	return []string{bittorrent.IPv4.String(), bittorrent.IPv6.String()}
+func addressFamilies() []bittorrent.AddressFamily {
+	return []bittorrent.AddressFamily{bittorrent.IPv4, bittorrent.IPv6}
 }
 
-func (ps *peerStore) leecherInfohashKey(af, ih string) string {
-	return af + "_L_" + ih
+// Constants used for constructing/deconstructing swarm keys.
+const (
+	swarmPrefix                = "swarm_"
+	seedersSuffix              = "_seeders"
+	leechersSuffix             = "_leechers"
+	leecherSwarmKeyLength      = len(swarmPrefix) + 20 + 1 + addressFamilyLength + len(leechersSuffix)
+	seederSwarmKeyLength       = len(swarmPrefix) + 20 + 1 + addressFamilyLength + len(seedersSuffix)
+	swarmKeyInfohashStart      = len(swarmPrefix)
+	swarmKeyInfohashEnd        = swarmKeyInfohashStart + 20
+	swarmKeyAddressFamilyStart = swarmKeyInfohashEnd + 1
+	swarmKeyAddressFamilyEnd   = swarmKeyAddressFamilyStart + addressFamilyLength
+)
+
+// The length of IPv4/IPv6.
+const addressFamilyLength = len(bittorrent.IPv4String)
```

Comment:
why not just use the byte/uint8? that seems like it'll save memory and a bunch of branches